vllm/config/__init__.py

```diff
@@ -2385,14 +2385,17 @@ def _verify_args(self) -> Self:
                              "speculative decoding is > 1, but got "
                              f"{self.disable_by_batch_size=}")
 
+        # Allow Qwen family variants (e.g., qwen, qwen2, qwen3) and Llama via
+        # a prefix match on the model family.
         eagle3_target_supported = ["llama", "qwen"]
-        if self.method == "eagle3" and self.target_model_config and not any(
-                supported_model in
-                self.target_model_config.hf_text_config.model_type
-                for supported_model in eagle3_target_supported):
-            raise ValueError(
-                f"Eagle3 is only supported for {eagle3_target_supported} models. "  # noqa: E501
-                f"Got {self.target_model_config.hf_text_config.model_type=}")
+        if self.method == "eagle3" and self.target_model_config:
+            model_type = str(
+                self.target_model_config.hf_text_config.model_type).lower()
+            if not any(model_type.startswith(supported)
+                       for supported in eagle3_target_supported):
```

Comment:
How is this different to the original behaviour? `model_type` is always a string. The only thing that looks different is the `lower()`.

This would be functionally the same:

```suggestion
        if self.method == "eagle3" and self.target_model_config and not any(
                supported_model in
                self.target_model_config.hf_text_config.model_type.lower()
                for supported_model in eagle3_target_supported):
```